comp/autoscaling/datadogclient/impl/client.go

```diff
@@ -130,3 +130,9 @@ func createDatadogClient(cfg configComponent.Component, logger logComp.Component
 	}
 	return newDatadogSingleClient(cfg, logger)
 }
+
+func (d *datadogClientWrapper) getNumberOfRefreshes() int {
```

Comment:
Just adding some clarification here for future reference: this is necessary because accessing `numberOfRefreshes` without a lock causes a data race. Since the only actual time `numberOfRefreshes` is [read in the code](https://github.com/DataDog/datadog-agent/blob/fbdd1bbd7ee4d151620c6a88c17fcbf1c632cf71/comp/autoscaling/datadogclient/impl/client.go#L115-L119) is already done with a lock, this function is only used in the unit test right now.

nit: could be added as a comment but if in the future we want to read `numberOfRefreshes` we should probably use this helper anyway (not sure if this is expected to change much though)